### DISCUSSION TITLE
somehow unstaged changes section doesn't show up at all

### DISCUSSION THREAD

**Discussion by wdkrnls** [magit/magit#4675] [Support] [answered]
I'm not sure if this is a bug or just some feature I acidently stumbled on to, but somehow magit doesn't show the unstaged changes section at all. Yet, when I examine `git status` I see quite a few entries. I don't see any errors. Did I trigger a feature to hide a section? All I see is `untracked files`, `staged changes`, and `unmerged into` in the main magit interface. I am using Magit 3.3.0 with Emacs 27.2 and git 2.35.1.

**Comment by tarsius** [comment, marked as answer]
Did you try in another repository?
Did you check if there is an error/warning *in the `*Messages*` buffer*?
Did you try to reproduce with `emacs -Q` (see https://magit.vc/manual/magit/Debugging-Tools.html)?

The relevant function is `magit-insert-unstaged-changes`. Try adding debug statements to that, among other things to find out whether it is being called at all.

**Reply by tarsius** [reply]
Try `git update-index --no-assume-unchanged`.

If that doesn't help, also (in a shell) stage those changes and then unstage them again?

**Reply by wdkrnls** [reply]
The `git update-index` command didn't do anything, but your latter suggestion helped reveal to me that git was barfing on a missing git extension: git-crypt. This was configured to watch over a few of the files with unstaged changes. I installed that extension and now I see the staged changes section in Magit again.

**Reply by tarsius** [reply]
I have added `M-x magit-toggle-git-debug` to make it easier to debug such issues.

**Comment by wdkrnls** [comment]
I did check for errors/warnings in the *Messages* buffer. There weren't any. I also ran shell command produced by `M-x magit-emacs-Q-command` against the same repository. The resulting Emacs Magit instance showed the same behavior against the repository I tested and reported above. I tried another couple of repositories, and did see unstaged changes show up in them: exactly the changes the git shell command shows there to be. So, it seems like Magit is getting confused about this one repository in particular. I so no differences in Magit behavior between my Emacs config and the Magit "emacs -Q" config.

**Comment by lafrenierejm** [comment]
I was encountering the same behavior. The unstaged section wasn't being displayed in one repo but was everywhere else. Enabling `magit-toggle-git-debug` and then checking the `*messages*` buffer showed what my specific issue was; I had a git filter configured to use [`git-crypt`](https://www.agwa.name/projects/git-crypt/) but didn't have `git-crypt` installed locally.
